cli/reporter/proto.go

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) Mondoo, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package reporter
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"google.golang.org/protobuf/encoding/protojson"
+	"strings"
+
+	"go.mondoo.com/cnquery/v10/explorer"
+	"go.mondoo.com/cnquery/v10/shared"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+func ConvertToProto(data *explorer.ReportCollection) (*Report, error) {
+	protoReport := &Report{
+		Assets: map[string]*Asset{},
+		Data:   map[string]*DataValues{},
+		Errors: map[string]string{},
+	}
+
+	if data == nil {
+		return protoReport, nil
+	}
+
+	queryMrnIdx := map[string]string{}
+
+	// this case can happen when all assets error out, eg. no query pack is available that matches
+	if data.Bundle != nil {
+		for i := range data.Bundle.Packs {
+			pack := data.Bundle.Packs[i]
+			for j := range pack.Queries {
+				query := pack.Queries[j]
+				queryMrnIdx[query.CodeId] = query.Mrn
+			}
+		}
+	}
+
+	// fill in assets
+	for assetMrn, a := range data.Assets {
+		pAsset := &Asset{
+			Mrn:  a.Mrn,
+			Name: a.Name,
```

Comment:
do we also want the platform name and type here?

Reply:
We do not have that information right now, in cnquery asset. We can add that later if required.